lib/views/home_content/feed/post_card/votes_widget.dart

```diff
@@ -1,43 +1,59 @@
 import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/vote/upvote_state.dart";
+import "package:proxima/viewmodels/upvote_view_model.dart";
 
 /// This widget is used to display the votes of a post.
 /// It contains two buttons to upvote and downvote the post and the number of votes.
-class VotesWidget extends StatelessWidget {
-  final int votes;
+class VotesWidget extends HookConsumerWidget {
+  final PostIdFirestore postId;
 
   const VotesWidget({
     super.key,
-    required this.votes,
+    required this.postId,
   });
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncPostVote = ref.watch(postVoteProvider((postId: postId)));
+    final votes = asyncPostVote.valueOrNull?.votes ?? 0;
```

Comment:
Interesting choice here to use `valueOrNull` instead of a circular value...
Are we sure this is the behavior we want ?

I don't have a definite answer as this is a case where it would maybe not be the best UX to have spinners on every vote count till the request resolves but in the other hand we have no error handling and this differs from our standard way to make async requests to DB in other parts of the project.

Reply:
To answer @gruvw question, I believe indeed that this is the behavior that we want.
In our application, we used a circular value at the places where there is only one element to wait for.
For instance, we have one circular value for the whole list of post.

Here, as you said we would need to have a circular value for each post. I believe this can be quite distracting for the user. Thus I believe that it is better to simply display "nothing" (no vote and 0 as number of votes) in the loading state.
Regarding error handling, I think it is better to silently ignore the errors as it allows the user to continue using the app without this feature (which isn't a absolute necessity to use the app).

If we want to provide an even smoother transition from the loading state to the data, I guess we could pass the initial number of votes as parameter to the widget (coming from the corresponding `PostOverview`). And we display this number of votes while the `UpVoteViewModel` is loading. But this is only an idea, what do you think ?

Reply:
To answer @CHOOSEIT question, we cannot use the `voteScore` coming from the `postOverview` to instanciate the `UpVoteViewModel`.
I also had this idea originaly but this leads to the following bug:

To limit ressources and db calls, the `UpVoteViewModel` are created on the fly only when the corresponding post is displayed on the screen. 
Thus, it can be created for two reasons:
- (1) The user manually refreshes the overview which rebuild the widget and so the view model
- (2) The user scroll down (thus the viewmodel is disposed) and then up again (thus the viewmodel is recreated)

These two possibilities implies that we cannot know for sure that the `voteScore` used to create the viewmodel is coherent with the database.
Indeed, consider the following scenario:
- The user opens the overview and see "postA" which has 10 votes.
- The user upvotes "postA".
- Thus, the viewmodel exposes locally a vote score of 11 but the `voteScore` from `postOverview` is still 10.
- The user scrolls down thus the viewmodel is disposed.
- Thue user scrolls back up .
- The viewmodel is re-instanciated with the `voteScore` of the `postOverview` which is 10 and it exposes this value.
- Thus we have an incoherent number of votes displayed.

This is why we need query the number of votes in the viewmodel. 
With that, we only need the `postId` as widget parameter.

Reply:
Maybe we could not have it as auto-dispose ?

Reply:
The issue if we not have it as auto-dispose is that the viewmodel won't be recreate when the user refreshes manually.
Thus, the number of votes wont refresh.